runtime/near-vm-logic/src/gas_counter.rs

```diff
@@ -95,56 +90,45 @@ impl GasCounter {
         }
     }
 
-    #[cfg(all(feature = "costs_counting", feature = "protocol_feature_evm"))]
+    #[cfg(feature = "protocol_feature_evm")]
     #[inline]
     pub fn inc_evm_gas_counter(&mut self, value: EvmGas) {
-        EVM_GAS_COUNTER.with(|f| {
-            *f.borrow_mut() += value;
-        })
+        if cfg!(all(feature = "costs_counting")) {
+            EVM_GAS_COUNTER.with(|f| {
+                *f.borrow_mut() += value;
+            })
+        }
     }
 
-    #[cfg(all(not(feature = "costs_counting"), feature = "protocol_feature_evm"))]
-    #[inline]
-    pub fn inc_evm_gas_counter(&mut self, _value: EvmGas) {}
-
-    #[cfg(feature = "costs_counting")]
     #[inline]
     fn inc_ext_costs_counter(&mut self, cost: ExtCosts, value: u64) {
-        EXT_COSTS_COUNTER.with(|f| {
-            *f.borrow_mut().entry(cost).or_default() += value;
-        });
+        if cfg!(feature = "costs_counting") {
+            EXT_COSTS_COUNTER.with(|f| {
+                *f.borrow_mut().entry(cost).or_default() += value;
+            })
+        }
     }
 
-    #[cfg(not(feature = "costs_counting"))]
-    #[inline]
-    fn inc_ext_costs_counter(&mut self, _cost: ExtCosts, _value: u64) {}
-
-    #[cfg(feature = "costs_counting")]
     #[inline]
     fn update_profile_host(&mut self, cost: ExtCosts, value: u64) {
-        match &self.profile {
-            Some(profile) => profile.add_ext_cost(cost, value),
-            None => {}
-        };
+        if cfg!(feature = "costs_counting") {
+            match &self.profile {
+                Some(profile) => profile.add_ext_cost(cost, value),
+                None => {}
+            }
+        }
     }
 
-    #[cfg(not(feature = "costs_counting"))]
-    #[inline]
-    fn update_profile_host(&mut self, cost: ExtCosts, _value: u64) {}
-
-    #[cfg(feature = "costs_counting")]
     #[inline]
     fn update_profile_action(&mut self, action: ActionCosts, value: u64) {
-        match &self.profile {
-            Some(profile) => profile.add_action_cost(action, value),
-            None => {}
-        };
+        if cfg!(feature = "costs_counting") {
+            match &self.profile {
```

Comment:
is it better to use `if let` here

Reply:
Yeah, let me refactor it as well while I am at it.

My rule of thumb is "use if let iff there's no else". With else, `match` actually reads better and I like that ignoring (`_`) is explicit.